workflow/sync/sync_manager.go

```diff
@@ -524,7 +524,9 @@ func (sm *Manager) ReleaseAll(wf *wfv1.Workflow) bool {
 	}
 
 	if wf.Status.Synchronization.Mutex != nil {
-		for _, holding := range wf.Status.Synchronization.Mutex.Holding {
+		h := make([]wfv1.MutexHolding, len(wf.Status.Synchronization.Mutex.Holding))
```

Comment:
Shouldn't this fix also be applied on line 499, for semaphores?

Reply:
Line 499 doesn't directly modify the `wf.Status.Synchronization.Semaphore.Holding` array, so there's no need to include a change for it. On the other hand, line 505 might cause issues if `holding.Holders` contains multiple elements.

That said, I'm not sure if there's a scenario where `holding.Holders` actually has more than one element. In all of my tests, it has always contained only one, and I've never encountered any related issues.

I haven’t seen a concrete issue yet, but would it be okay if I bring it up again later if I do? I haven’t found a relevant issue to include in that PR for now.

Reply:
Ok, can we just add something like the test I've suggested to prove the equivalent works.

Reply:
Thanks